pkg/skaffold/build/dependencies.go

```diff
@@ -66,3 +81,20 @@ func DependenciesForArtifact(ctx context.Context, a *latest.Artifact, cfg docker
 
 	return util.AbsolutePaths(a.Workspace, paths), nil
 }
+
+// CreateBuildArgsFromArtifacts creates docker build args for an artifact from its required artifacts slice.
+func CreateBuildArgsFromArtifacts(deps []*latest.ArtifactDependency, r ArtifactResolver) (map[string]*string, error) {
+	if r == nil {
+		// `diagnose` is called without an artifact resolver. Return an empty map in this case.
+		return nil, nil
+	}
+	m := make(map[string]*string)
+	for _, d := range deps {
+		t, err := r.GetImageTag(d.ImageName)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Would this be simpler if it was more like a map and returned a `(tagged string, found bool)`?  It never feels right to ignore error returns, even when you have a reasonable case (like you make above) — and artifact-not-found doesn't seem like an actual error case.  I can't really imagine what errors should come from an ArtifactResolver since we're not intending this to do remote lookups.

Reply:
that makes a lot of sense. Will refactor.

Reply:
done.